torchreid/losses/metric.py

```diff
@@ -301,7 +301,7 @@ def init_iteration(self):
     def end_iteration(self):
         """Finalizes a training iteration"""
 
-        self.last_loss_value.backward(retain_graph=True)
+        self.last_loss_value.backward()
```

Comment:
It looks like the gradients have been already calculated during the stage: `total_loss.backward(retain_graph=self.enable_metric_losses)`.

Reply:
Made this optional it.